wallet/import.go

```diff
@@ -1,13 +1,232 @@
+	}
+}
+
+// validateExtendedPubKey ensures a sane derived public key is provided.
+func (w *Wallet) validateExtendedPubKey(pubKey *hdkeychain.ExtendedKey,
+	isAccountKey bool) error {
+
+	// Private keys are not allowed.
+	if pubKey.IsPrivate() {
+		return errors.New("private keys cannot be imported")
+	}
+
+	// The public key must have a version corresponding to the current
+	// chain.
+	if !w.isPubKeyForNet(pubKey) {
+		return fmt.Errorf("expected extended public key for current "+
+			"network %v", w.chainParams.Name)
+	}
+
+	// Verify the extended public key's depth and child index based on
+	// whether it's an account key or not.
+	if isAccountKey {
+		if pubKey.Depth() != accountPubKeyDepth {
+			return errors.New("invalid account key, must be of the " +
+				"form m/purpose'/coin_type'/account'")
+		}
+		if pubKey.ChildIndex() < hdkeychain.HardenedKeyStart {
+			return errors.New("invalid account key, must be hardened")
+		}
+	} else {
+		if pubKey.Depth() != pubKeyDepth {
+			return errors.New("invalid account key, must be of the " +
+				"form m/purpose'/coin_type'/account'/change/" +
+				"address_index")
+		}
+		if pubKey.ChildIndex() >= hdkeychain.HardenedKeyStart {
+			return errors.New("invalid pulic key, must not be " +
+				"hardened")
+		}
+	}
+
+	return nil
+}
+
+// ImportAccount imports an account backed by an account extended public key.
+// The master key fingerprint denotes the fingerprint of the root key
+// corresponding to the account public key (also known as the key with
+// derivation path m/). This may be required by some hardware wallets for proper
+// identification and signing. The address type can usually be inferred from the
+// key's version, but in the case of legacy versions (xpub, tpub), an address
+// type must be specified as we intend to not support importing BIP-44 keys into
+// the wallet using the legacy pay-to-pubkey-hash (P2PKH) scheme.
+func (w *Wallet) ImportAccount(name string, accountPubKey *hdkeychain.ExtendedKey,
+	masterKeyFingerprint uint32, addrType *waddrmgr.AddressType) (
+	waddrmgr.KeyScope, uint32, error) {
+
+	// Ensure we have a valid account public key.
+	if err := w.validateExtendedPubKey(accountPubKey, true); err != nil {
+		return waddrmgr.KeyScope{}, 0, err
+	}
+
+	// Determine what key scope the account public key should belong to and
+	// init the account as watch-only.
+	keyScope, err := keyScopeFromPubKey(accountPubKey, addrType)
+	if err != nil {
+		return waddrmgr.KeyScope{}, 0, err
+	}
+	scopedMgr, err := w.Manager.FetchScopedKeyManager(keyScope)
+	if err != nil {
+		return waddrmgr.KeyScope{}, 0, err
+	}
+
+	var account uint32
+	err = walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+		ns := tx.ReadWriteBucket(waddrmgrNamespaceKey)
+		account, err = scopedMgr.NewAccountWatchingOnly(
+			ns, name, accountPubKey, masterKeyFingerprint,
+		)
+		return err
+	})
+	if err != nil {
+		return waddrmgr.KeyScope{}, 0, err
+	}
+
+	return keyScope, account, nil
```

Comment:
Perhaps specify that what's being returned here is the _internal_ (auto incremented) account? Alternatively, we could retuyrn the full account info struct here as well.

Reply:
Returned the full account info struct instead.